src/deploy/functions/validate.ts

```diff
@@ -85,3 +79,26 @@ export function packageJsonIsValid(
     throw new FirebaseError(msg);
   }
 }
+
+function _functionsSourceIsPresent(data: any, sourceDir: string, projectDir: string): void {
```

Comment:
not necessary (in typescript) to prefix "private" functions with `_`. the fact that they're not exported is enough.

Reply:
Done.